### ISSUE TITLE
Large keyboard input delay, Windows 10

### ISSUE THREAD

**Issue by ethanx94** [stefansundin/superf4#1]
When using the dev preview builds of Windows 10 with this application running, at times there is a huge delay in keyboard input system wide. Quitting superf4 or restarting my computer is the only way to regain a usable input speed.
This doesn't happen every time the superf4 is running, I've noticed it triggering at very strange times when doing things involving switching video output or heavy gpu usage.
For example, after quitting a demanding game full screen exclusive game like GTA V it will start.
Another occurrence is watching full screen exclusive video through Pot Player media player.
As for switching video output, when I trigger the windows project settings with Win+P and switch between these options (PC Screen only to Extend for instance) this bug starts happening then as well.
If it makes any difference, I am using a Corsair K70 RGB keyboard with a Utility process running in the background, however I've noticed this problem with a generic Dell keyboard too. Unplugging and replugging either keyboard makes no difference. Superf4 must be closed to regain functionality.

**Comment by stefansundin** [comment]
I intend to test Win 10 when it is released. Hopefully I can solve this issue then. Thanks!

**Comment by benlen10** [comment]
I second this issue. For the past few days I have been experiencing serious input lag and have tried absolutely everything to resolve it. I finally ended up reinstalling Windows 10 fresh and managed to narrow the issue down the Super F4. It is an amazing program but causes serious problems with Windows 10.

**Comment by stefansundin** [comment]
I have tested SuperF4 on Win10 now, and I haven't encountered any problems.

I just released a new version, because google code is shutting down. I doubt it will fix the problem though.

Feel free to play around with the source code, maybe you two can fix it. Check out the build instructions for my other program, AltDrag, to help you with compiling SuperF4. It should be very similar. https://stefansundin.github.io/altdrag/doc/build.html

Post here if you have problems and I will try to help you.

**Comment by chrisdok** [comment]
I experience a 5-30 second keyboard delay on Windows 10 when opening certain games while running Super F4. This doesn't happen always, but can always be fixed by closing Super F4. The game consistently causing the problem now, is Metal Gear Solid 5 for me, but I remember having the same issues before.

**Comment by jyrkive** [comment]
This randomly happens to me as well. I have SuperF4 start automatically along with my computer (in a fairly complex way: USBDLM launches PStart, which in turn starts SuperF4). At some days (something like once per month) I have the keyboard input lag described here, and it goes away if I terminate SuperF4. Also, when that happens, the SuperF4 tray icon doesn't react to right-clicking (so I have to use Task Manager to kill it). Like the others, I use Windows 10.

**Comment by stefansundin** [comment]
That sounds similar to some problems I had when trying to start SuperF4 via the task scheduler. This was needed before I added the administrator elevation code in v1.3. I think it worked a lot better if I made the task scheduler wait 30 seconds before starting SuperF4.

Any chance the task scheduler is involved in your setup?

**Comment by jyrkive** [comment]
Well, it's possible. I don't know how, exactly, USBDLM launches programs. Its documentation states:
> Since Vista it's quite tricky to start a program from a service and get its window activated. If this is not required at all then please configure
> `openstyle=noactivate`

(And I haven't configured that because PStart is a program launcher, and it's useful to have it activated.)

**Comment by Kornstalx** [comment]
Did this issue ever get resolved? I've used SuperF4 for years and almost can't live without it, but under Windows 10 it's definitely causing input lag when elevated. Killing the SuperF4 process via task manager then relaunching it is the only way to remedy the problem, but it happens so often it's aggravating.

**Comment by stefansundin** [comment]
Sorry, I was never able to reproduce this. I'd love to fix it if I could.

**Comment by sam2795** [comment]
I want to start this off by saying thank you for SuperF4, I've used it basically since it came out and it's been an essential program I use daily, but it recently started the massive amount of keyboard delay that other people have been experiencing. Its specifically happening when I launch Overwatch and Civ5 which is strange because  it has never happened before with either of those games.

**Comment by ProGamerzFTW** [comment]
I've had it happen two times so far after launching Trove and Payday 2. They were both fullscreen aswell. It may be random when it occurs.

**Comment by stefansundin** [comment]
The issue #4 was filed a few days ago, and it gave me an idea. I made a custom version that doesn't even contain the keyboard hook. It only uses the "TimerCheck" workaround I made for certain games. Try it out and see if it works better. Go here to download the zip: https://github.com/stefansundin/superf4/issues/4#issuecomment-246912571

**Comment by stefansundin** [comment]
Hi @ethanx94, @benlen10, @chrisdok, @jyrkive, @Kornstalx, @sam2795, @ProGamerzFTW and everyone else who have experienced this problem.

I have made a new build that includes the fix by vmrob in #5. Can you try it and see if it fixes the problem?

File: [SuperF4-no-block.zip](https://github.com/stefansundin/superf4/files/679513/SuperF4-no-block.zip)

Sorry for the long delay in looking into this. All credit goes to vmrob for finding the problem.

If this proves to solve the problem everyone is experiencing, then I will release a new official version with the fix.

Thank you!

**Comment by jyrkive** [comment]
@stefansundin I haven't had the input delay problem at all for months, so I can't really test if the test build fixes it.

**Comment by tasger22** [comment]
Thank you, for me it has been fixed. The issue on my end was caused by higher than native resolution, now the lag hasn't occured. Hope it fixes the problem for everybody else.

**Comment by WilliamFeely** [comment]
Hi.  I had been having the same problem, which had been triggered whenever I switch to a different DPI scaling on my monitor.  First I suspected the Splashtop Streamer which wasn't working for me.  No change.
Then I stumbled my way into task manager and killed SuperF4.  Instant relief on my keyboard.

**Comment by luke1408** [comment]
I have to second this issue. It seems to trigger when I awake my notebook from sleep mode. Restarting Super F4 fixes it tho.

**Comment by jyrkive** [comment]
The input delay problem returned for me when I got a 4K monitor. My system isn't powerful enough for 4K gaming: therefore I play at 1080p, which changes system DPI setting from 150% to 100%. And that triggered the input delay problem every time.

And the important part: The issue is **fixed** in the test build. Thank you, @vmrob ! 👍

**Comment by stefansundin** [comment]
Ouch gubelfax. I hope you also needed a new PC for other reasons, but it does make me feel a bit responsible.

Can you try the build posted [a few comments above](https://github.com/stefansundin/superf4/issues/1#issuecomment-269853636)?

**Comment by lollitroll** [comment]
god...it took me a week to find the problem.
Thx for the fix though; works for me!

**Comment by tommy61157** [comment]
Having the Same issue oddly ever since I got my UHD monitor, never had an issue before then. Downloaded and ran the application, hopefully this will fix it, but until there's a 1.4 patch, definitely not going to set SuperF4 to run by default for awhile, barely used it anyways

**Comment by Jadelor** [comment]
This fixed version corrects the problem. Thanks a lot – might want to make it available on the official page.

**Comment by ethanx94** [comment]
I opened this issue three years ago and I stayed subscribed so I’m still getting notifications to this day. But I’m happy people are finding and fixing their problems. I think that’s just magical. 😂

**Comment by LockeColeFF** [comment]
I am having this issue as well.  To recreate it reliably, you must have Super F4 running and then launch an application that changes the resolution of your screen.  In my case it was games that entered fullscreen mode.  Only fix I found was to disable Super F4.  If a fix could get added to the binary that would be wonderful because I friggin love this program!

**Comment by SirWindfield** [comment]
Might be the new windows update...

**Comment by MobinWeekend** [comment]
Ok, I found out. A game I lunched was set on 4k by default on first run (my windows is in 1080 res) and that caused it.
Every game that I run in 4K will cause keyboard input lags. and TimerCheck won't fix it.

**Comment by Mirellanmawl** [comment]
Oh god finally. As others here I had a really hard time narrowing down the problem. I re-installed Windows (well, the laptop had also gotten infected with malware, so it was not that big of a deal) and was about to hand it in and try to get my computer fixed through my warranty. But it seemed so much like to a software error so what would they do? They would just reset it and the problem would still be there. After at least an hour of searching the web and trying different things, I found this thread.

The error occured when I plugged in or unplugged my second monitor that had a different DPI setting (I did not figure out DPI mattered until I read the comments here). And sometimes seemingly just randomly. Turning on the TimerCheck option did not fix it. But with the build you posted here I can happily say it works.

Here is my system information:
https://hastebin.com/iboganilun.cs

**Comment by stefansundin** [comment]
Hey everyone. I am very sorry for ignoring everyone for all these years. I really am, and I apologize.

I released a new version with the fix: https://github.com/stefansundin/superf4/releases/tag/v1.4

**Comment by dxgldotorg** [comment]
Heads-up, your github.io site still says 1.3 and links to the 1.3 installer.

**Comment by stefansundin** [comment]
I just updated it two minutes ago. :)

**Comment by SuperKael** [comment]
Wow, just found this. Huh. Today happens to be my birthday, and I have been struggling with this all day due to this acting up while my PC is plugged into our TV. Great to find the fix. Better late than never, I suppose :P

**Comment by thebigsmileXD** [comment]
Magic wonders, i was stumbling across this problem just some minutes ago and finally found this issue, being the first reasonable solution - thanks for this issue to exist, good to know that 1.4 has a fix, i was still running 1.3. I will test the new version from now on, Greetings from germany :)

**Comment by Fangh** [comment]
I had this problem too today and I did not know there were a V1.4 ! Thank you !

**Comment by Truandale** [comment]
> I second this issue. For the past few days I have been experiencing serious input lag and have tried absolutely everything to resolve it. I finally ended up reinstalling Windows 10 fresh and managed to narrow the issue down the Super F4. It is an amazing program but causes serious problems with Windows 10.

Thanks for the advice. When I uninstalled SuperF4, the problem disappeared.
